tsdb/record/record.go

```diff
@@ -42,6 +43,69 @@ const (
 	Tombstones Type = 3
 	// Exemplars is used to match WAL records of type Exemplars.
 	Exemplars Type = 4
+	// Metadata is used to match WAL records of type Metadata.
+	Metadata Type = 5
```

Comment:
Just realised one thing. We (@jesusvazquez and I) plan to upstream the out-of-order support which is already using the type 5. It will be hard to change it to something else there since it is already under use (not via Prometheus). Can we change it to 6 here? (We will upstream out of order code as soon as possible to avoid such conflicts in future)

Reply:
Done, thanks for the catch, it would have been ugly to sort out 😅